orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -0,0 +1,501 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/logging/logging.h"
+#include "core/common/logging/sinks/clog_sink.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/graph/graph_viewer.h"
+#include "core/graph/model.h"
+#include "core/platform/path_lib.h"
+#include "core/platform/env.h"
+#include "core/providers/cpu/cpu_execution_provider.h"
+#include "core/util/protobuf_parsing_utils.h"
+#include "onnx/defs/tensor_proto_util.h"
+#include "orttraining/core/framework/checkpointing.h"
+#include "orttraining/core/framework/protobuf_message_sequence.h"
+#include "orttraining/training_api/checkpoint.h"
+
+namespace onnxruntime {
+namespace training {
+namespace api_test {
+
+namespace {
+
+constexpr const char* k_tensor_proto_file_name = ORT_TSTR("tensors.pbseq");
+constexpr const char* k_external_data_file_name = ORT_TSTR("tensors.bin");
+constexpr const char* k_tensor_proto_properties_file_name = ORT_TSTR("properties.pbseq");
+
+constexpr const char* k_param_root_prefix = "param";
+constexpr const char* k_optimizer_root_prefix = "optim";
+constexpr const char* k_property_root_prefix = "custom";
+constexpr const char* k_name_seperator = "_";
+
+PathString GetTensorProtoFilePath(const PathString& checkpoint_directory, const std::string& filename_prefix) {
+  return ConcatPathComponent<PathChar>(checkpoint_directory, ORT_TSTR(filename_prefix + k_name_seperator) + k_tensor_proto_file_name);
+}
+
+PathString GetExternalDataFilePath(const PathString& checkpoint_directory, const std::string& filename_prefix) {
+  return ConcatPathComponent<PathChar>(checkpoint_directory, ORT_TSTR(filename_prefix + k_name_seperator) + k_external_data_file_name);
+}
+
+PathString GetTensorProtoPropertiesFilePath(const PathString& checkpoint_directory, const std::string& filename_prefix) {
```

Comment:
GetTensorProtoFilePath,  GetExternalDataFilePath, GetTensorProtoPropertiesFilePath are all doing almost the same thing. I think if you just keep one util mtd for this and add the file name as another param then you can get rid of so many mtds.

Reply:
As talked offline, those are only needed by checkpoint.cc, so better to put them here without public accessibility. Have moved some that are used by others into orttraining/orttraining/core/framework/checkpoint_common.h.

Reply:
Removed GetExternalDataFilePath.